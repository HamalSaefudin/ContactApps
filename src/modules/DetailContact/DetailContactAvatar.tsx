import Icon, {ICON_CATEGORY} from '@/components/Icons';
import Colors from '@/theme/Colors';
import Layout from '@/theme/Layout';
import React from 'react';
import {Image, StyleSheet, TouchableOpacity, View} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import ShimmerPlaceholder from 'react-native-shimmer-placeholder';
import {moderateScale} from 'react-native-size-matters';

type Props = {
  isLoading: boolean;
  onPress: () => void;
  source: string;
};

function DetailContactAvatar({isLoading, onPress, source}: Props) {
  return (
    <ShimmerPlaceholder
      visible={!isLoading}
      LinearGradient={LinearGradient}
      shimmerStyle={styles.contentImg}>
      <TouchableOpacity
        testID="contact-avatar-btn"
        onPress={onPress}
        style={styles.contentImg}>
        <Image
          testID="contact-avatar-image"
          source={{
            uri: source,
          }}
          style={styles.img}
        />
        <View style={styles.contentImgCreate}>
          <Icon
            testID="contact-avatar-icon"
            category={ICON_CATEGORY.FONTISTO}
            name="spinner-refresh"
          />
        </View>
      </TouchableOpacity>
    </ShimmerPlaceholder>
  );
}
const styles = StyleSheet.create({
  contentImg: {
    height: moderateScale(150),
    width: moderateScale(150),
    borderRadius: moderateScale(75),
    backgroundColor: Colors.common.border,
    marginBottom: moderateScale(16),
    ...Layout.justifyContentCenter,
    ...Layout.alignItemsCenter,
  },
  contentImgCreate: {
    height: moderateScale(25),
    width: moderateScale(25),
    borderRadius: moderateScale(25),
    backgroundColor: Colors.common.border,
    ...Layout.justifyContentCenter,
    ...Layout.alignItemsCenter,
    position: 'absolute',
    right: moderateScale(20),
    bottom: 0,
  },
  img: {
    ...Layout.fullSize,
    borderRadius: moderateScale(75),
  },
});

export default DetailContactAvatar;
