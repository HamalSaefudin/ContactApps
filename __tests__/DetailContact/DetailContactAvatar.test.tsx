import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import DetailContactAvatar from '@/modules/DetailContact/DetailContactAvatar';

describe('DetailContactAvatar', () => {
  const mockSource = 'https://example.com/photo.jpg';
  const mockOnPress = jest.fn();

  it('to match snapshot', () => {
    const {toJSON} = render(
      <DetailContactAvatar
        isLoading={false}
        onPress={mockOnPress}
        source={mockSource}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when not loading', () => {
    const {getByTestId} = render(
      <DetailContactAvatar
        isLoading={false}
        onPress={mockOnPress}
        source={mockSource}
      />,
    );

    const image = getByTestId('contact-avatar-image');
    expect(image.props.source.uri).toBe(mockSource);

    const icon = getByTestId('contact-avatar-icon');
    expect(icon).toBeTruthy();
  });

  it('calls onPress when pressed', () => {
    const {getByTestId} = render(
      <DetailContactAvatar
        isLoading={false}
        onPress={mockOnPress}
        source={mockSource}
      />,
    );

    fireEvent.press(getByTestId('contact-avatar-btn'));
    expect(mockOnPress).toHaveBeenCalled();
  });
});
