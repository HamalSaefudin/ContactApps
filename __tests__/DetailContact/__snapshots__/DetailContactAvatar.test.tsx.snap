// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailContactAvatar to match snapshot 1`] = `
<View
  style={
    [
      false,
      {
        "overflow": "hidden",
      },
      false,
      undefined,
    ]
  }
>
  <View
    style={
      [
        false,
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#D9D9D9",
          "borderRadius": 117.85714285714286,
          "height": 235.71428571428572,
          "justifyContent": "center",
          "marginBottom": 25.142857142857142,
          "opacity": 1,
          "width": 235.71428571428572,
        }
      }
      testID="contact-avatar-btn"
    >
      <Image
        source={
          {
            "uri": "https://example.com/photo.jpg",
          }
        }
        style={
          {
            "borderRadius": 117.85714285714286,
            "height": "100%",
            "width": "100%",
          }
        }
        testID="contact-avatar-image"
      />
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#D9D9D9",
            "borderRadius": 39.285714285714285,
            "bottom": 0,
            "height": 39.285714285714285,
            "justifyContent": "center",
            "position": "absolute",
            "right": 31.428571428571427,
            "width": 39.285714285714285,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontSize": 25.142857142857142,
              },
              undefined,
              {
                "fontFamily": "Fontisto",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
          testID="contact-avatar-icon"
        >
          î§¸
        </Text>
      </View>
    </View>
  </View>
</View>
`;
